lib/matplotlib/colorbar.py

```diff
@@ -1114,19 +1114,17 @@ def _locate(self, x):
             xn = self.norm(x, clip=False).filled()
 
         # The rest is linear interpolation with extrapolation at ends.
-        ii = np.searchsorted(b, xn)
-        i0 = ii - 1
-        itop = (ii == len(b))
-        ibot = (ii == 0)
-        i0[itop] -= 1
-        ii[itop] -= 1
-        i0[ibot] += 1
-        ii[ibot] += 1
-
-        y = self._y
-        db = b[ii] - b[i0]
-        dy = y[ii] - y[i0]
-        z = y[i0] + (xn - b[i0]) * dy / db
+
+        bunique = b
+        yunique = self._y
+        if b[0] == b[1]:
```

Comment:
I see how the previous code was doing linear interpolation by hand, but where do these "equal first and second point / equal last and nexttolast point" cases come from?

Reply:
Fixed comment:
```
        bunique = b
        yunique = self._y
        # trim extra b values at beginning and end if they are
        # not unique.  These are here for extended colorbars, and are not
        # wanted for the interpolation.  
```

Reply:
I don't quite understand this. Souldn't you trim all but one duplicate values?
The current code does `b = [0, 0, 0, 1, 2] --> bunique = [0, 0, 1, 2]`. Is that intended?

Reply:
That should never happen - if it does, just as well that an error trips.  `b` are the boundaries, passed through the norm should be monotonic, except for, possibly, the first and last values if there are an "extend" either at max or min.